hadoop-project/pom.xml

```diff
@@ -168,7 +168,18 @@
     <enforced.maven.version>[3.3.0,)</enforced.maven.version>
 
     <!-- Plugin versions and config -->
-    <maven-surefire-plugin.argLine>-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError</maven-surefire-plugin.argLine>
+    <maven-surefire-plugin.argLine>-Xmx2048m -XX:+HeapDumpOnOutOfMemoryError --add-exports=jdk.compiler/com.sun.tools.javac.util=ALL-UNNAMED
```

Comment:
Thank you very much for submitting this PR. We are looking forward to Hadoop supporting JDK11 and JDK17.

I have some questions:

Should we fully support JDK11 and JDK17, or should we try to avoid certain issues by adding some compilation parameters?

From my perspective, we should find ways to upgrade the code and make all modules fully compatible with the recommended syntax of JDK11 and JDK17.

@aajisaka @ayushtkn @steveloughran

Reply:
it's a big ongoing bit of work; aws have been busier than the rest of us...we do test against later java versions, but do the build/UT/IT against v8.

I'd like to move 3.5 and then 3.4.x to java17+ ..., where x is a release which comes out this year but is greater than 1